save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt

```diff
@@ -82,6 +87,37 @@ internal class OrganizationController(private val organizationService: Organizat
         return Mono.just(response)
     }
 
+    /**
+     * @param organization updateOrganization
+     * @param authentication an [Authentication] representing an authenticated request
+     * @return response
+     */
+    @PostMapping("/{organizationName}/update")
+    @PreAuthorize("isAuthenticated()")
+    fun updateOrganization(@RequestBody organization: Organization, authentication: Authentication): Mono<StringResponse> {
+        val userId = (authentication.details as AuthenticationDetails).id
+        val role = lnkUserOrganizationService.findRoleByUserIdAndOrganizationName(userId, organization.name)
+        val response = if (role.priority >= 2) {
+            organizationService.updateOrganization(organization)
+            ResponseEntity.ok("Organization updated")
+        } else {
+            ResponseEntity.badRequest().body("Access denied")
```

Comment:
Returning `400` here is too general. Since all organizations are public, you can return `403` here.

Reply:
Fixed